mmcls/models/backbones/convnext.py

```diff
@@ -0,0 +1,292 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from itertools import chain
+from typing import Sequence
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.cnn.bricks import DropPath
+from mmcv.runner import BaseModule
+from mmcv.runner.base_module import ModuleList, Sequential
+
+from ..builder import BACKBONES
+from .base_backbone import BaseBackbone
+
+
+class LayerNorm2d(nn.LayerNorm):
+    """LayerNorm that supports two data formats: channels_last (default) or
+    channels_first.
+
+    The ordering of the dimensions in the inputs. channels_last corresponds to
+    inputs with shape (batch_size, height, width, channels) while
+    channels_first corresponds to inputs with shape (batch_size, channels,
+    height, width).
+    """
+
+    def __init__(self,
+                 num_channels: int,
+                 dim: int = 1,
+                 eps: float = 1e-6,
+                 **kwargs) -> None:
+        super().__init__(num_channels, eps=eps, **kwargs)
+        self.num_channels = self.normalized_shape[0]
+        self.dim = dim
+
+    def forward(self, x):
+        if self.dim == -1 or self.dim == x.dim() - 1:
```

Comment:
it is necessary to explain arg `dim` or add a comment here.

Reply:
Done.